src/cdk/v2/destinations/http/procWorkflow.yaml

```diff
@@ -55,13 +56,12 @@ steps:
   - name: prepareBody
     template: |
       const payload = $.getCustomMappings(.message, .destination.Config.propertiesMapping);
-      $.context.payload = $.excludeMappedFields(payload, .destination.Config.propertiesMapping)
-      $.context.format === "XML" &&  !$.isEmptyObject($.context.payload) ?  $.context.payload = {payload: $.getXMLPayload($.context.payload)} : $.context.payload = $.removeUndefinedAndNullValues($.context.payload)
+      $.context.payload = $.prepareBody(payload, $.context.format);
 
   - name: buildResponseForProcessTransformation
     template: |
       const response = $.defaultRequestConfig();
-      $.context.format === "JSON" ? response.body.JSON = $.context.payload: response.body.XML = $.context.payload;
+      response.body[$.context.format] = $.context.payload;
```

Comment:
shouldn't there be `.` after body ? But I don't see any test cases failing 
      response.body.[$.context.format] = $.context.payload;

Reply:
No KC, `response.body[$.context.format]` is the right syntax